### PR TITLE
Rework verification state sync per latest proto schema.

### PR DESCRIPTION
**Pull request by charlesmchen**
Can't test this until desktop is ready.

PTAL @michaelkirk

### CODE REVIEW

File: src/Devices/OWSVerificationStateSyncMessage.m [hunk 1]
```diff
@@ -66,26 +66,41 @@ - (OWSSignalServiceProtosSyncMessage *)buildSyncMessage
...
 }
 
+- (NSData *)createRandomNSDataOfSize:(size_t)size
+{
+    NSMutableData *data = [NSMutableData dataWithCapacity:size];
+    for (size_t i = 0; i < size; ++i) {
+        u_int32_t randomBits = arc4random_uniform(256);
+        [data appendBytes:(void *)&randomBits length:1];
+    }
+    return data;
```
Comment (michaelkirk) [inline src/Devices/OWSVerificationStateSyncMessage.m]: Can you use `[Cryptography generateRandomBytes:size]` instead?

File: src/Messages/OWSIdentityManager.m [hunk 2]
```diff
@@ -608,6 +615,10 @@ - (void)tryToApplyVerificationStateFromSyncMessage:(OWSVerificationState)verific
         OWSFail(@"Verification state sync message missing identityKey: %@", recipientId);
         return;
     }
+    
+    if (![self isSyncEnabled]) {
+        return;
```
Comment (michaelkirk) [inline src/Messages/OWSIdentityManager.m]: nit: log?

File: src/Messages/OWSIdentityManager.m [hunk 3]
```diff
@@ -513,16 +513,23 @@ - (void)syncAllVerificationStates
...
 {
     OWSAssert(message);
     OWSAssert(message.recipientIds.count > 0);
 
-    if (YES) {
-        // Don't actually transmit any verification state sync messages
-        // until we finalize the proto schema changes.
-        //
-        // TODO: Remove.
+    if (![self isSyncEnabled]) {
```
Comment (michaelkirk) [inline src/Messages/OWSIdentityManager.m:532]: nit: log?

File: protobuf/OWSSignalServiceProtos.proto [hunk 4]
```diff
@@ -124,26 +124,26 @@ message SyncMessage {
...
 
-  message Verification {
+  message Verified {
     enum State {
-      DEFAULT            = 0;
-      VERIFIED           = 1;
-      NO_LONGER_VERIFIED = 2;
+      DEFAULT    = 0;
+      VERIFIED   = 1;
+      UNVERIFIED = 2;
```
Comment (michaelkirk) [inline protobuf/OWSSignalServiceProtos.proto:131]: 😞

**Review by michaelkirk — APPROVED** [review]
LGTM

assuming desktop is following the same protos, not our prior?

File: src/Messages/OWSIdentityManager.m (hunk 2 above)
Comment (charlesmchen) [reply src/Messages/OWSIdentityManager.m]: Done. We're reverting this in a day or two though.

File: src/Messages/OWSIdentityManager.m (hunk 3 above)
Comment (charlesmchen) [reply src/Messages/OWSIdentityManager.m:532]: Done. We're reverting this in a day or two though.

File: src/Devices/OWSVerificationStateSyncMessage.m (hunk 1 above)
Comment (charlesmchen) [reply src/Devices/OWSVerificationStateSyncMessage.m]: Done.

**Comment by charlesmchen** [comment]
> assuming desktop is following the same protos, not our prior?

They need to converge on Android/service proto schema if they haven't already.

**Pull request merged**